nimble/host/src/ble_hs_mbuf.c

```diff
@@ -84,15 +85,33 @@ ble_hs_mbuf_l2cap_pkt(void)
     return ble_hs_mbuf_gen_pkt(BLE_HCI_DATA_HDR_SZ + BLE_L2CAP_HDR_SZ);
 }
 
+struct os_mbuf *
+ble_hs_mbuf_hdr_pkt(void)
+{
+    struct os_mbuf *om;
+    uint16_t leading_space = BLE_HS_ATT_PKT_HDR_LEADINGSPACE;
+
+
+    /* This is just for header which is going to be added later on, that is
+    * why size is equal to leading_space
+    */
+    om = os_msys_get_pkthdr(leading_space, leading_space);
```

Comment:
I must be missing something, because I still don't understand why we want a user-header length equal to `leading_space`.  Why not specify 0 here, and then perform `om->om_data += leading_space`?

Specifying `leading_space` as the user-header length causes `om->om_pkthdr_len` to be equal to `sizeof (struct os_mbuf_pkthdr) + `leading_space`.  Does the code ever decrease `om_pkthdr_len`?  If not, I think this space can never be used.

Reply:
you are right, I misread a code. So there is no API to actually add leading space to mbuf contains packet header other then adding it manually.